burn-wgpu/src/template/mask/where_inplace.wgsl

```diff
@@ -33,12 +39,20 @@ fn main(@builtin(global_invocation_id) global_id: vec3<u32>) {
         let shape_value = info[i + 4u * dim];
         let shape_mask = info[i + 5u * dim];
 
-        index_input += global_id.x / stride_input % shape_input * stride_input;
-        index_value += global_id.x / stride_input % shape_value * stride_value;
-        index_mask += global_id.x / stride_input % shape_mask * stride_mask;
+        index_input += id / stride_input % shape_input * stride_input;
+        index_value += id / stride_input % shape_value * stride_value;
+        index_mask += id / stride_input % shape_mask * stride_mask;
+    }
+
+    var condition = mask[index_mask] != 0u;
```

Comment:
technically, lines 47-51 could be simplified to 
`let condition = mask[index] != reverse;`
But I'm not sure the performance gain would outweigh the readability loss

Reply:
Nop, since you can only compare to 0s, not 1s, since the buffer cast from float to u32 can transform any non zero value to anything else, you can only do comparison with zero.